### DISCUSSION TITLE
FieldResolvers in generated type such as Aggregate

### DISCUSSION THREAD

**Discussion by ghost** [MichalLytek/typegraphql-prisma#308] [Q&A] [answered]
Hello,

Is there a way to include FieldResolvers in the generated resolver types such as `Aggregate<X>Resolver` or `GroupBy<X>Resolver`?

Thanks!

**Comment by MichalLytek** [comment, marked as answer]
What do you mean by include field resolvers?
You want to query for relation when getting results from aggregate and group?

**Reply by ghost** [reply]
Hi @MichalLytek ,

Yes correct. Below is a rough example:


```ts
@Resolver(of => User)
export class CustomUserResolver {
  @FieldResolver(type => String)
  async favoriteColor(
    @Root() user: User,
  ): Promise<string> {
    return "Red"
  }
}
```


```gql

query groupuser {
  groupByUser(by: ['favoriteColor']) {
     favoriteColor
}
}

```

**Reply by MichalLytek** [reply]
Ohh, I think your example is not possible. Prisma group/aggregates by db columns, not some computed on-the-fly virtual fields.

**Reply by ghost** [reply]
Thanks!
